### PR TITLE
fix(admin-vm): surface real errors from InitializeMint and handle JIT races

### PR DESCRIPTION
**Pull request by Huzaifa696**
## Summary

The Admin VM previously accepted any `InitializeMint`, including re-inits that silently overwrote live mints. This PR validates the instruction, refuses re-init, and tightens the operator JIT path so the races the VM now rejects recover via an on-chain re-check.

- **Admin VM**: validates SPL `InitializeMint` wire format, returns `AccountAlreadyInitialized` on live mints, fails atomically.
- **Router**: mixed admin/non-admin txs go to the real SVM.
- **Operator JIT**: gates on `is_initialized_mint_data`; on a non-`Confirmed` result, re-reads the mint on-chain with backoff and treats an already-initialised mint as success.

## Files changed

| File | Change |
|------|--------|
| `core/src/vm/admin.rs` | InitializeMint validation, re-init guard, atomic failure |
| `core/src/stages/execution.rs` | Admin/regular/mixed tx routing |
| `indexer/src/operator/sender/mint.rs` | Fast-path + JIT race recovery with retried on-chain re-check |

## Tests

- Admin VM: 13 tests covering happy path, all validation failures, re-init refusal, multi-instruction atomicity.
- Router: 8 tests covering pure-admin / mixed / regular partitioning and worker lifecycle.
- Operator: 6 tests for `is_initialized_mint_data` corner cases.

End-to-end verified via `/tmp/test_admin_vm.py` against the running stack, refer [ticket](https://linear.app/solana-fndn/issue/EXO-20/write-node-admin-vm-always-returns-ok). The integration test `test_with_redis` exercises the back-to-back deposit race.

## Impact

Admin VM returns accurate statuses instead of false positives. Operators no longer stall on a JIT re-init race when the read-RPC hasn't caught up yet.

### Coverage Report

| Component | Lines Hit | Lines Total | Coverage | Artifact |
|-----------|-----------|-------------|----------|----------|
| Core | 8,287 | 9,726 | 85.2% | [rust-unit-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673059) |
| Indexer | 13,525 | 15,800 | 85.6% | [rust-unit-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673059) |
| Gateway | 952 | 1,076 | 88.5% | [rust-unit-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673059) |
| Auth | 541 | 596 | 90.8% | [rust-unit-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673059) |
| Withdraw Program | 118 | 230 | 51.3% | [unit-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673064) |
| Escrow Program | 1,170 | 1,951 | 60.0% | [unit-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673064) |
| E2E Integration | 8,080 | 11,934 | 67.7% | [e2e-coverage-reports](https://github.com/solana-foundation/contra/actions/runs/24719673059) |
| **Total** | **32,673** | **41,313** | **79.1%** | |

> Last updated: 2026-04-21 11:45:04 UTC by E2E Integration

### CODE REVIEW

File: core/src/vm/admin.rs [hunk 1]
```diff
@@ -166,6 +186,78 @@ impl AdminVm {
...
+    /// All byte indices below refer to this layout.
+    fn process_initialize_mint<CB: TransactionProcessingCallback>(
+        callbacks: &CB,
+        tx: &impl SVMTransaction,
+        instruction: solana_svm_transaction::instruction::SVMInstruction,
+    ) -> Result<(Pubkey, AccountSharedData), InstructionError> {
+        // Minimum payload: instruction must reference the mint as an account,
+        // and data must span discriminator + decimals + mint_authority (up to
+        // byte 34). The COption tag at byte 34 is inspected separately below.
+        if instruction.accounts.is_empty() || instruction.data.len() < 34 {
```
[the remark this answers is not in the document]
Comment (Huzaifa696) [reply core/src/vm/admin.rs]: Agreed, tightening to 35 matches spl-token's wire format

**Review by dev-jodee — APPROVED** [review]
